public/src/components/admin_button_objs.jsx

```diff
@@ -0,0 +1,67 @@
+export default {
+  view_edit(props) {
+    const { thisPage } = props;
+    let target;
+    if (thisPage === 'Question') {
```

Comment:
The other components are stateless, so I figured it made sense to put this logic in the parent.